### ISSUE TITLE
Use a autoformatting tool (integrate in pipeline too)

### ISSUE THREAD

**Issue by maiksensi** [codecentric/chaos-monkey-spring-boot#133]
In order to avoid "nitpicking" comments about the styling of a java file I suggest we integrate an autoformatting tool into our maven goals (which can be executed then) and also integrate that as a mandatory goal into our CI pipeline (if not correctly formatted fail pipeline).
In the best case, we only look at pure syntax formatting (like prettier does for JavaScript).

**Comment by WtfJoke** [comment]
I suggest we should using [google-java-format](https://github.com/google/google-java-format) and using either this [revelc/formatter-maven-plugin](https://github.com/revelc/formatter-maven-plugin) (configurable maven format plugin) or this [coveooss/fmt-maven-plugin](https://github.com/coveooss/fmt-maven-plugin) (google style format maven plugin)

**Comment by maiksensi** [comment]
As long as it works in the pipeline (checker) and in let's say intellij I really dont care as long as we get a consistent style by adding this.

**Comment by WtfJoke** [comment]
https://github.com/diffplug/spotless looks neat and they also have a gradle/maven plugin. It supports google java format out of the box.

Seen in habitcentric (gradle):
```yaml
spotless {
    java {
        googleJavaFormat('1.7')
    }
}
```

**Comment by fletchgqc** [comment]
I used google java format with fmt-maven-plugin on a recent project and it was a very positive experience. Although it takes a while to get used to the annoyance of your build being declared broken, it's fantastic to have a standard format which is enforced during the build chain. Diffs are much easier to read, etc.

I believe that google format is the right choice because it's good enough (actually very good) and by not thinking about it, we avoid having to have some kind of format selection process.

I have no experience with the other maven plugins. Happy to go with whatever others choose.

**Comment by maiksensi** [comment]
I am happy to prepare a MR in the next days (already started tbh) including: 
- setup the plugin with checkstyle and a supression file (for disabled rules)
- setup running in the pipeline and failing it when violations occur
- docs explaining how to setup intellij 

Docs: 
https://google.github.io/styleguide/javaguide.html

They have an intellij plugin: 
https://raw.githubusercontent.com/google/styleguide/gh-pages/intellij-java-google-style.xml

IntelliJ Code Style File: 
https://raw.githubusercontent.com/google/styleguide/gh-pages/intellij-java-google-style.xml

**Comment by WtfJoke** [comment]
Closed by #160
